jbmc/src/java_bytecode/java_static_initializers.cpp

```diff
@@ -817,6 +817,13 @@ code_blockt get_user_specified_clinit_body(
           }
         }
       }
+      if(symbol_table.lookup(real_clinit_name) == nullptr)
+      {
+        // Case where there is a user provided clinit, but the real clinit
+        // doesn't appear in the symbol table. This may occur when some class
```

Comment:
@antlechner 
`create_user_specified_clinit_function_symbol` is called if `needs_clinit_wrapper` is true, which is when `clinit` exists *or* some other condition. If you think this is wrong then `needs_clinit_wrapper` needs to be corrected.

Reply:
Ah, that's the part I overlooked! `needs_clinit_wrapper` is true if `clinit` exists for this class _or_ a superclass.
So it looks like the mistake is that currently, we create a `user_specified_clinit` iff we create a `clinit_wrapper`. But really we should be creating a `user_specified_clinit` iff there is a `clinit`.

Reply:
ok, I'm still going to merge this PR since it prevents a crash but the condition for calling `create_user_specified_clinit_function_symbol` should be fixed. Do you want to create an issue for that?